RecoParticleFlow/PFProducer/interface/PFTables.h

```diff
@@ -0,0 +1,194 @@
+#include "DataFormats/ParticleFlowReco/interface/PFRecHitFraction.h"
+#include "DataFormats/ParticleFlowReco/interface/PFCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementSuperCluster.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementGsfTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementBrem.h"
+#include "DataFormats/ParticleFlowReco/interface/PFBlockElementTrack.h"
+#include "RecoParticleFlow/PFProducer/interface/TableDefinitions.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+namespace edm::soa {
+  TrackTableVertex makeTrackTableVertex(std::vector<reco::PFBlockElement*> const& targetSet);
+
+  TrackTableExtrapolation makeTrackTable(std::vector<reco::PFBlockElement*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  TrackTableExtrapolation makeTrackTable(std::vector<const reco::PFBlockElementGsfTrack*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  TrackTableExtrapolation makeTrackTable(std::vector<const reco::PFBlockElementBrem*> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+  template <class RecTrackType>
+  TrackTableExtrapolation makeTrackTable(std::vector<RecTrackType> const& objects,
+                                         reco::PFTrajectoryPoint::LayerType layerType);
+
+  ConvRefTable makeConvRefTable(const std::vector<reco::ConversionRef>& convrefs);
+  ConvBremTable makeConvBremTable(const std::vector<reco::PFRecTrackRef>& convbrems);
+  BremTable makeBremTable(const std::vector<const reco::PFBlockElementBrem*>& brems);
+
+  RecHitTable makeRecHitTable(std::vector<const reco::PFRecHitFraction*> const& objects);
+  SuperClusterRecHitTable makeSuperClusterRecHitTable(std::vector<const std::pair<DetId, float>*> const& objects);
+  ClusterTable makeClusterTable(std::vector<const reco::PFBlockElementCluster*> const& objects);
+  SuperClusterTable makeSuperClusterTable(std::vector<const reco::PFBlockElementSuperCluster*> const& objects);
+  GSFTable makeGSFTable(std::vector<const reco::PFBlockElementGsfTrack*> const& objects);
+}  // namespace edm::soa
+
+//this is a transient data structure, made in order to collect the data that PFBlockAlgo needs
+//holds all input data to PFBlockAlgo computations
+
+template <class ClusterTable, class RecHitTable>
+class PFClusterTables {
+public:
+  ClusterTable cluster_table_;
+  RecHitTable rechit_table_;
+  std::vector<size_t> cluster_to_element_;
+  std::vector<size_t> element_to_cluster_;
+  std::unordered_map<size_t, std::set<size_t>> rechit_to_cluster_;
+  std::unordered_map<size_t, std::set<size_t>> cluster_to_rechit_;
+
+  PFClusterTables() = default;
+  PFClusterTables(ClusterTable&& _cluster_table,
```

Comment:
leading underscores should be avoided, rule 2.14

Reply:
Also applies to _linkdata.

Reply:
thanks